Sources/ProjectDescription/GraphDefinition.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+/// A styling definition to be used when elements are rendered in graphs
+public struct GraphDefinition: Hashable, Equatable, CustomStringConvertible, CustomDebugStringConvertible, Codable {
+    /// The shape of the object
+    public enum Shape: String, Hashable, Codable {
+        case box, rectangle, square, circle, ellipse, point, egg, triangle, plaintext, plain, diamond, trapezium, parallelogram,
+             house, hexagon, octagon, doublecircle, doubleoctagon, invtriangle, invtrapezium, invhouse, Mdiamond, Msquare,
+             Mcircle,
+             polygon, oval, star, cylinder, note, tab, folder, box3d, component, cds, signature
+    }
+
+    /// The color of the object. Can either be an hex string like "#FFCC00" or a color name like "black", "teal", etc.
```

Comment:
Is there a list of valid colors, should we provide an enum?

Reply:
It's a choice that comes with some tradeoffs.
Such list (like the shapes) comes directly from GraphViz, is passed down to it and parsed by the library itself.
From what I've understood, the library accepts a variety of color types, from rgb (hex string) to hls to color name ("aquamarine" etc).
In my opinion, people who will use this feature will probably extend the Color object providing some static vars with hex strings and use them in code (like you would do in normal app coding, you enum all the hardcoded values).

Having the burden of maintaining a complex list of colors is really hard and probably not worth it. 

Also please keep in mind that GraphViz for Swift (tuist dependency) is an archived repository since 2021, and may be a candidate for replacement in favor of something else.